packages/manager/apps/web/client/app/hosting/database/order/components/private-database/private-database.controller.js

```diff
@@ -0,0 +1,168 @@
+    };
+
+    this.loadingDependencies = true;
+    this.$q
+      .all({
+        services: this.OrderPrivateDatabaseService.getServices(),
+        dependencies: this.loadDependencies(),
+      })
+      .then(({ services }) => {
+        this.services = services;
+      })
+      .catch((error) => {
+        this.handleError(error);
+      })
+      .finally(() => {
+        this.loadingDependencies = false;
+      });
+  }
+
+  loadDependencies() {
+    return this.OrderPrivateDatabaseService.getCatalog(this.user.ovhSubsidiary)
+      .then((catalog) => {
+        this.catalog = catalog;
+        this.productOffers.workflowOptions.catalog = this.catalog;
+        return this.OrderPrivateDatabaseService.getProducts(catalog);
+      })
+      .then((products) => {
+        this.products = products;
+        this.catalogProducts = this.OrderPrivateDatabaseService.constructor.getCatalogProducts(
+          this.catalog,
+          products,
+        );
+        return true;
+      });
+  }
+
+  getVersions(product) {
+    this.catalogProduct = find(this.catalogProducts, {
+      planCode: product.planCode,
+    });
+    const engines = find(this.catalogProduct.configurations, {
+      name: 'engine',
+    });
+
+    if (!engines || !engines.values) {
+      return [];
+    }
+
+    return orderBy(
+      engines.values.sort().map((engine) => {
+        const [engineLabel, engineVersion] = engine.split('_');
+        const name = this.$translate.instant(
+          `hosting_database_order_private_version_${engineLabel}`,
+          { version: engineVersion },
+        );
+
+        return {
+          name,
+          value: engine,
+          engineLabel,
+          engineVersion: parseFloat(engineVersion),
+        };
+      }),
+      ['engineLabel', 'engineVersion'],
+      ['asc', 'desc'],
+    );
+  }
+
+  onServiceChange(serviceName) {
+    this.loadingDatacenter = true;
+    this.order.datacenter = null;
+    this.getDatacenter(serviceName)
+      .then((datacenter) => {
+        this.order.datacenter = datacenter;
+      })
+      .catch((error) => {
+        this.handleError(error);
+      })
+      .finally(() => {
+        this.loadingDatacenter = false;
+      });
+  }
+
+  onProductChange(product) {
+    this.product = product;
+    this.versions = this.getVersions(product);
+  }
+
+  getDatacenter(serviceName) {
+    return this.OrderPrivateDatabaseService.getDatacenter(serviceName);
+  }
+
+  cancel() {
+    if (this.onCancel && isFunction(this.onCancel)) {
+      this.onCancel();
+    }
+  }
+
+  handleSuccess(order) {
+    this.$window.open(order.url, '_blank');
+    if (this.onSuccess && isFunction(this.onSuccess)) {
+      this.onSuccess({ url: order.url });
+    }
+  }
+
+  handleError(error) {
+    if (this.onError && isFunction(this.onError)) {
+      this.onError({ error });
+    }
+  }
+
+  getPlanCode() {
+    return this.product.planCode;
+  }
+
+  getConfiguration() {
+    const conf = [
+      {
+        label: 'dc',
```

Comment:
Does moving these labels to a constant file make sense?

Reply:
Its only used in one place. I keep them as it is.